### PR TITLE
Removed the sTMHMMoves array and made TMs/HMs read moves from their secondaryId item field

### PR DESCRIPTION
**Pull request by LOuroboros**
## Description
About 2 weeks ago, I brought up a suggestion inside the Discord server; to remove the pointless `sTMHMMoves` array that linked the `ITEM_TM` and `ITEM_HM` constants to the `MOVE` constants of the moves that those TMs and HMs teach InGame.
Since no one else seemed to be up for it and I'm bored, I decided to do it myself.

Apart from removing that array, I also deleted the never used `registrability` field present in the items' data.
I did that in order to preserve the size of `gItems`, as I had to upgrade the `secondaryId` field to a u16 to account for all the moves past `MOVE_SPIT_UP`.

## **Discord contact info**
Lunos#4026

### CODE REVIEW

**Comment by LOuroboros** [comment]
There's an extra change that I wanted to make, but that I didn't do.
I'd like to hear opinions before doing anything.

Basically, I wanted to make `ItemIdToBattleMoveId` return `MOVE_NONE` if the item that it currently scans is not a TM/HM.
```diff
u16 ItemIdToBattleMoveId(u16 item)
{
-   return sTMHMMoves[tmNumber];
+   return (ItemId_GetPocket(item) == POCKET_TM_HM) ? gItems[item].secondaryId : MOVE_NONE;
}
```
The purpose of this change would be to protect against user errors.
If a user passed through this function an item like `ITEM_POTION`, it would return `MOVE_NONE`, where as if someone passes the `ITEM_TM50` instead, the function returns `MOVE_OVERHEAT`.

The thing is, I'm not sure how useful that would be since it would all be rather silent, and the user would have to be aware of this change at all for it to be any effective.

On a side note, I just realized that I broke `IsMoveHm`. I'll fix it in a moment.

**Comment by LOuroboros** [comment]
Alright, it's ready for a review. I'll leave it as a draft until I get opinions about the change to `ItemIdToBattleMoveId`.

File: include/pokemon.h [hunk 1]
```diff
@@ -579,7 +579,7 @@ const u32 *GetMonFrontSpritePal(struct Pokemon *mon);
 const u32 *GetMonSpritePalFromSpeciesAndPersonality(u16 species, u32 otId, u32 personality);
 const struct CompressedSpritePalette *GetMonSpritePalStruct(struct Pokemon *mon);
 const struct CompressedSpritePalette *GetMonSpritePalStructFromOtIdPersonality(u16 species, u32 otId , u32 personality);
-bool32 IsHMMove2(u16 move);
+bool8 IsMoveHm(u16 move);
```
Comment (mrgriffin) [inline include/pokemon.h:582]: I'd probably call this `IsMoveHM` with a capital M, but admittedly we use both across the codebase (:sob:).

EDIT: In fact, `IsMoveHm` already existed and you just moved it. ffs! :joy:

**Review by mrgriffin — COMMENTED** [review]
Huh, I didn't realize that the expanded TMs 51-100 don't have moves by default!

In any case, I think this is a nice improvement! I'd also like to see the "extra change" you mentioned in your comment. I think it's helpful even if the user isn't aware, because currently if they pass `ITEM_POTION` they get some arbitrary value back—whatever happens to be in some particular address in the ROM—consistently getting `MOVE_NONE` back should make it easier for somebody to realize they're passing things that aren't TMs into the function :)

File: include/pokemon.h (hunk 1 above)
Comment (LOuroboros) [reply include/pokemon.h:582]: Yeah, haha. I removed the original `IsMoveHm` and turned `IsHMMove2` into the new `IsMoveHm` :eyes:

I'll rename `IsMoveHm` into `IsMoveHM`, I don't mind and it *is* consistent with its old label too *(`IsHMMove2`)*, so I'll push :eyes:

**Pull request merged**